projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,591 @@
+- Feature Name: nano-contract-integration
+- Start Date: 2023-12-13
+- RFC PR: https://github.com/HathorNetwork/rfcs/pull/72
+- Hathor Issue:
+- Author: Alex Ruzenhack alex@hathor.network
+
+# Summary
+[summary]: #summary
+
+This design aims to provides for users the ability to read Nano Contract transactions, its state, and the related blueprint details. In addition, this design also provides for users the ability to sign Nano Contract transactions using a dapp with Wallet Connect integration.
+
+# Motivation
+[motivation]: #motivation
+
+The user gains the ability to consult and interact with Nano Contract in a portable way, using the wallet-mobile, but also gains the ability to extend its usage by connecting to dapps using Wallet Connect.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## UI Design Guidance
+
+In this document, you will find sections titled **Design Bill** that outline requirements for UI screens and components. These sections provide guidelines while allowing designers room for creativity. A key principle to guide your work is **Clear Sign.** This principle emphasizes that information presented to users should be crafted with clear and knowledgeable intent. For more details, please refer to the article ["Clear-Sign Your Worries Away."](https://www.ledger.com/blog/clear-sign-your-worries-away)
+
+## Unleash
+[unleash]: #unleash
+
+First, we should enable the user's device in the `nano-contract.rollout` feature flag on Unleash. For this feature flag we use the default variant with 3 strategies:
+1. UserIds: this first strategy is dedicated for internal testers
+2. UserIds: this second strategy is dedicated for external testers
```

Comment:
In the code, nothing changes. They are logical groups. I'm doing this to increase our control and convenience. Sometimes we want to turn-off the feature for everybody except for ourselves and by having these groups segregated we can achieve this with more precision and less work, it would be a no brainer.